administrator/components/com_media/models/api.php

```diff
@@ -38,85 +46,128 @@ public function __construct($config = array())
 	{
 		parent::__construct($config);
 
+		if (!isset($config['providers']))
+		{
+			$config['providers'] = JPluginHelper::getPlugin('filesystem');
+			$this->providers = $config['providers'];
+		}
+
 		if (!isset($config['fileadapter']))
 		{
-			// Import Local file system plugin
+			// Import enabled file system plugins
 			JPluginHelper::importPlugin('filesystem');
 
 			$app = JFactory::getApplication();
 
 			$results = $app->triggerEvent('onFileSystemGetAdapters');
+			$adapters = array();
 
-			if ($results != null)
+			for ($i = 0; $i < count($results); $i++)
```

Comment:
if it's in the second part of the for then count() is calculated in every iteration ;)